File: app/src/main/java/com/mxkj/htmusic/mymodule/fragment/RecruitPeopleAllFragment.java
//package com.mxkj.haitang.mymodule.fragment;
//
//import android.content.Intent;
//import android.os.Bundle;
//import android.support.annotation.Nullable;
//import android.support.v7.widget.LinearLayoutManager;
//import android.support.v7.widget.RecyclerView;
//import android.util.Log;
//import android.view.LayoutInflater;
//import android.view.View;
//import android.view.ViewGroup;
//import android.widget.TextView;
//
//import com.alibaba.fastjson.JSON;
//import com.mxkj.haitang.R;
//import com.mxkj.haitang.mymodule.adapter.ParticipateFragmentAdapter;
//import com.mxkj.haitang.messagemodule.bean.ParticipateFragmentBean;
//import com.mxkj.haitang.mymodule.adapter.RecruitPeopleAllFragmentAdapter;
//import com.mxkj.haitang.mymodule.adapter.RecruitPeopleParticipateFragmentAdapter;
//import com.mxkj.haitang.projectmodule.bean.ProjectdetailsListBean;
//import com.mxkj.yuanyintang.net.NetWork;
//import com.trello.rxlifecycle2.components.support.RxFragment;
//
//import java.util.ArrayList;
//
//import okhttp3.Headers;
//
//public class RecruitPeopleAllFragment extends RxFragment {
//    private View rootView;
//
//    private RecyclerView recyclerview;
//    private RecruitPeopleAllFragmentAdapter musicListMusicIanAdapter  = null;
//    private ArrayList<ParticipateFragmentBean.DataBean> musicListMusicIanBeanList= new ArrayList<>();
//
//    private TextView tv_no_data;
//
//
//    String work_type_id="";
//    String require_id = "";
//
//
//
//    @Nullable
//    @Override
//    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
//        if (rootView == null) {
//            rootView = inflater.inflate(R.layout.participatefragment, container, false);
//        }
//        ViewGroup parent = (ViewGroup) rootView.getParent();
//        if (parent != null) {
//            parent.removeView(rootView);
//        }
//        initEvent();
//        return rootView;
//    }
//
//    private void initEvent() {
//
//        recyclerview = rootView.findViewById(R.id.recyclerview);
//        tv_no_data = rootView.findViewById(R.id.tv_no_data);
//
//
//        Bundle bundle = getArguments();
//        if(null!=bundle.getString("require_id")) {
//            work_type_id = bundle.getString("work_type_id");
//            require_id = bundle.getString("require_id");
//        }
//
//        initdata();
//    }
//    private void initdata() {
//        //需要请求来tinachognshuju
//        ParticipateDate();
//
//        recyclerview.setLayoutManager(new LinearLayoutManager(getActivity()));
//        if (musicListMusicIanAdapter == null) {
//
//            if(musicListMusicIanBeanList.size()<1){
//                tv_no_data.setVisibility(View.VISIBLE);
//                tv_no_data.setText("你还没有选定该工种的音乐人呢");
//            }else {
//                tv_no_data.setVisibility(View.GONE);
//            }
//
//            musicListMusicIanAdapter = new RecruitPeopleAllFragmentAdapter(musicListMusicIanBeanList);
//            recyclerview.setAdapter(musicListMusicIanAdapter);
//        } else {
//            musicListMusicIanAdapter.setNewData(musicListMusicIanBeanList);
//        }
//    }
//
//    private void ParticipateDate() {
//        ParticipateFragmentBean.DataBean dataBean1 = new ParticipateFragmentBean.DataBean();
//        dataBean1.setTitle("一白二穷工作室");
//        dataBean1.setHead("0");
//        ParticipateFragmentBean.DataBean dataBean2 = new ParticipateFragmentBean.DataBean();
//        dataBean2.setTitle("MortonWill莫顿");
//        dataBean2.setHead("1");
//        ParticipateFragmentBean.DataBean dataBean3 = new ParticipateFragmentBean.DataBean();
//        dataBean3.setTitle("机智过人的阿怜酱");
//        dataBean3.setHead("4");
//        ParticipateFragmentBean.DataBean dataBean4 = new ParticipateFragmentBean.DataBean();
//        dataBean4.setTitle("叶爱ryeowook");
//        dataBean4.setHead("2");
//        ParticipateFragmentBean.DataBean dataBean5 = new ParticipateFragmentBean.DataBean();
//        dataBean5.setTitle("夜雨寄北古风团队夜雨寄...");
//        dataBean5.setHead("3");
//        musicListMusicIanBeanList.add(dataBean1);
//        musicListMusicIanBeanList.add(dataBean2);
//        musicListMusicIanBeanList.add(dataBean3);
//        musicListMusicIanBeanList.add(dataBean4);
//        musicListMusicIanBeanList.add(dataBean5);
//
//    }
//}

File: app/src/main/java/com/mxkj/htmusic/mymodule/fragment/RecruitPeopleAllFragment.kt
package com.mxkj.htmusic.mymodule.fragment

import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.alibaba.fastjson.JSON
import com.mxkj.htmusic.R
import com.mxkj.htmusic.mymodule.adapter.RecruitPeopleAllFragmentAdapter
import com.mxkj.htmusic.projectmodule.bean.SelectedMusicansBean
import com.mxkj.yuanyintang.net.NetWork
import com.trello.rxlifecycle2.components.support.RxFragment
import okhttp3.Headers
import java.util.ArrayList

class RecruitPeopleAllFragment : RxFragment() {
    private var page: Int = 1
    private var rootView: View? = null
    private var recyclerview: RecyclerView? = null
    private var musicListMusicIanAdapter: RecruitPeopleAllFragmentAdapter? = null
    private val musicListMusicIanBeanList = ArrayList<SelectedMusicansBean.DataBean>()
    private var tv_no_data: TextView? = null


    internal var work_type_id: String? = ""
    internal var require_id: String? = ""
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        if (rootView != null) {
            val parent = rootView!!.parent as ViewGroup
            parent?.removeView(rootView)
        } else {
            rootView = inflater.inflate(R.layout.participatefragment, container, false)
        }
        initEvent()
        return rootView
    }

    private fun initEvent() {
        recyclerview = rootView!!.findViewById<RecyclerView>(R.id.recyclerview)
        tv_no_data = rootView!!.findViewById<TextView>(R.id.tv_no_data)

        val bundle = arguments
        if (null != bundle!!.getString("require_id")) {
            work_type_id = bundle.getString("work_type_id")
            require_id = bundle.getString("require_id")
        }
        initdata()
    }

    private fun initdata() {
        //需要请求来tinachognshuju
        ParticipateDate()

        recyclerview!!.setLayoutManager(LinearLayoutManager(activity))
        if (musicListMusicIanAdapter == null) {

            if (musicListMusicIanBeanList.size < 0) {
                tv_no_data!!.setVisibility(View.VISIBLE)
                tv_no_data!!.setText("你还没有选定该工种的音乐人呢")
            } else {
                tv_no_data!!.setVisibility(View.GONE)
            }

            musicListMusicIanAdapter = RecruitPeopleAllFragmentAdapter(musicListMusicIanBeanList)
            recyclerview!!.setAdapter(musicListMusicIanAdapter)
        } else {
            musicListMusicIanAdapter!!.setNewData(musicListMusicIanBeanList)
        }
    }

    private fun ParticipateDate() {

        NetWork.getcontractlist(activity!!, "", "","", Integer.parseInt(require_id), object : NetWork.TokenCallBack {
            override fun doNext(resultData: String, headers: Headers?) {
                jsonPondData(resultData)
            }

            override fun doError(msg: String) {

            }

            override fun doResult() {}
        })

//        val dataBean1 = SelectedMusicansBean.DataBean()
//        val dataBean11 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean11.project_title = "一白二穷工作室"
//        dataBean1.require_status = 0
//        val dataBean2 = SelectedMusicansBean.DataBean()
//        val dataBean22 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean22.project_title = "MortonWill莫顿"
//        dataBean2.require_status = 1
//        val dataBean3 = SelectedMusicansBean.DataBean()
//        val dataBean33 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean33.project_title = "机智过人的阿怜酱"
//        dataBean3.require_status = 2
//        val dataBean4 = SelectedMusicansBean.DataBean()
//        val dataBean44 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean44.project_title = "叶爱ryeowook"
//        dataBean4.require_status = 3
//        val dataBean5 = SelectedMusicansBean.DataBean()
//        val dataBean55 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean55.project_title = "夜雨寄北古风团队夜雨寄..."
//        dataBean5.require_status = 4
//
//        val dataBean6 = SelectedMusicansBean.DataBean()
//        val dataBean66 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean66.project_title = "一白二穷工作室"
//        dataBean6.require_status = 5
//        val dataBean7 = SelectedMusicansBean.DataBean()
//        val dataBean77 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean77.project_title = "MortonWill莫顿"
//        dataBean7.require_status = 6
//        val dataBean8 = SelectedMusicansBean.DataBean()
//        val dataBean88 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean88.project_title = "机智过人的阿怜酱"
//        dataBean8.require_status = 7
//        val dataBean9 = SelectedMusicansBean.DataBean()
//        val dataBean99 = SelectedMusicansBean.DataBean.ProjectBean()
//        dataBean99.project_title = "叶爱ryeowook"
//        dataBean9.require_status = 8
//
//
//        dataBean1.project = dataBean11
//        dataBean2.project = dataBean22
//        dataBean3.project = dataBean33
//        dataBean4.project = dataBean44
//        dataBean5.project = dataBean55
//        dataBean6.project = dataBean66
//        dataBean7.project = dataBean77
//        dataBean8.project = dataBean88
//
//        musicListMusicIanBeanList.add(dataBean1)
//        musicListMusicIanBeanList.add(dataBean2)
//        musicListMusicIanBeanList.add(dataBean3)
//        musicListMusicIanBeanList.add(dataBean4)
//        musicListMusicIanBeanList.add(dataBean5)
//        musicListMusicIanBeanList.add(dataBean6)
//        musicListMusicIanBeanList.add(dataBean7)
//        musicListMusicIanBeanList.add(dataBean8)
//        musicListMusicIanAdapter = RecruitPeopleAllFragmentAdapter(musicListMusicIanBeanList)
//        recyclerview!!.setAdapter(musicListMusicIanAdapter)
    }

    private fun jsonPondData(resultData: String) {
//        interRefresh?.setStopRefreshing()
        val selectedMusicansBean = JSON.parseObject(resultData, SelectedMusicansBean::class.java)
        val jsonArray = selectedMusicansBean?.data
        if (jsonArray!!.size > 0) {
//            rootView?.view_nodata?.visibility = View.GONE
            val toJSONString = JSON.toJSONString(jsonArray)
            val parse = JSON.parseArray(toJSONString, SelectedMusicansBean.DataBean::class.java)
            if (page == 1) {
                musicListMusicIanBeanList.clear()
            }
            musicListMusicIanBeanList.addAll(parse)
            musicListMusicIanAdapter?.setNewData(musicListMusicIanBeanList)
        } else if (page == 1) {
//            rootView?.view_nodata?.visibility = View.VISIBLE
        }
    }

}